QHyper/problems/TSP.py

```diff
@@ -0,0 +1,135 @@
+import itertools
+
+import numpy as np
+
+from .problem import Problem
+
+
+class TSP:
+    """TSP class"""
+
+    def __init__(self, number_of_cities: int, coords_range: tuple[float, float]=(0, 10000)) -> None:
+        self.number_of_cities = number_of_cities
+        self.coords_range = coords_range
+        self.cities_coords = self.get_cities()
+        self.distance_matrix = self.calculate_distance_matrix()
+        self.normalized_distance_matrix = self.normalize_distance_matrix()
+    
+    def get_cities(self) -> list[tuple[float, float]]:
+        cities_coords = np.random.randint(self.coords_range[0], self.coords_range[1], size = (self.number_of_cities, 2))
+        return cities_coords
+           
+    def normalize_cities(self) -> list[tuple[float, float]]:
+        max_coords = np.amax(self.cities_coords, axis=0)
+        normalized_cities_coords = np.divide(self.cities_coords, max_coords)
+        return normalized_cities_coords
+
+    def calculate_distance_between_points(
+        self, point_A: tuple[float, float], point_B: tuple[float, float]) -> float:
+        return np.sqrt((point_A[0] - point_B[0])**2 + (point_A[1] - point_B[1])**2)
+    
+    def calculate_distance_matrix(self) -> list[list[float]]:
+        distance_matrix = np.zeros((self.number_of_cities, self.number_of_cities))
+        for i in range(self.number_of_cities):
```

Comment:
[I know this is the code I've written] Do you think using enumerate() instead of range would be more readable in cases like this?

Reply:
noo, I much more prefer `range` than `enumerate`, especially it's range over number of cities which makes sense.